components/net/fetch/methods.rs

```diff
@@ -163,8 +188,15 @@ pub fn main_fetch(
         }
     }
 
+    // Step 2.2.
+    if should_request_be_blocked_by_csp(request) == csp::CheckResult::Blocked {
+        response = Some(Response::network_error(NetworkError::Internal(
+            "Blocked by Content-Security-Policy".into(),
+        )))
+    }
```

Comment:
AFAICT that's actually step 2.4., step 2.2. being about reporting the violations, not setting response to a network error.